### PR TITLE
[WIP]feat: alert context

### PR DESCRIPTION
**Pull request by SarthakC**
## PR Checklist
Please check that your PR fulfills the following requirements:

- [x] The commit message follows [our guidelines](https://github.com/pesto-students/batch-10-tripen/wiki/Commit-message-guidelines)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] GIFs/Videos/Images added (compulsory for frontend changes)

## PR Type
What kind of change does this PR introduce? 

- [ ] Bugfix
- [x] Feature
- [ ] Tests changed / added / updated
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Infrastructure changes
- [ ] Dependency upgrade
- [ ] Other... Please describe:

## Issue Number: 
#72 

## Does this PR introduce a breaking change?
- [ ] Yes
- [ ] No

## Other information

### CODE REVIEW

**Review by karansapolia — APPROVED** [review]
Recommend always including a gif of the functionality. Would appreciate one for this PR too :slightly_smiling_face:

**Review by p-racoon — APPROVED** [review]
need KT on this

File: client/src/context/alert/AlertState.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import alertReducer from './alertReducer';
+
+import { SET_ALERT, REMOVE_ALERT } from '../types.js';
+
+const AlertState = (props) => {
+  const initialState = [];
+  const [state, dispatch] = useReducer(alertReducer, initialState);
+
+  // Set Alert
+  const setAlert = (msg, type, timeout = 5000) => {
```
Comment (rishabh1403) [inline client/src/context/alert/AlertState.js]: nice :)

File: client/src/context/auth/AuthState.js [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  };
+  const [state, dispatch] = useReducer(authReducer, initialState);
+
+  // Load User
+  const loadUser = async () => {
+    if (localStorage.token) {
+      setAuthToken(localStorage.token);
+    }
+    try {
+      const res = await axios.get('/api/auth');
```
Comment (rishabh1403) [inline client/src/context/auth/AuthState.js]: this api request will not work if client and server are hosted cross domain.
have a constant with server url and make a request accordingly :)

File: client/src/context/auth/AuthState.js [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+      dispatch({ type: AUTH_ERROR });
+    }
+  };
+  // Register User
+  const register = async (formData) => {
+    const config = {
+      headers: { 'Content-Type': 'application/json' },
+    };
+    try {
+      const res = await axios.post('/api/users', formData, config);
```
Comment (rishabh1403) [inline client/src/context/auth/AuthState.js]: same as above... cross domain

File: client/src/context/auth/AuthState.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+      dispatch({
+        type: REGISTER_SUCCESS,
+        payload: res.data,
+      });
+      loadUser();
+    } catch (err) {
+      dispatch({
+        type: REGISTER_FAIL,
+        payload: err.response.data.msg,
```
Comment (rishabh1403) [inline client/src/context/auth/AuthState.js]: what if err doesn't have response...response doesn't have data and data doesn't have mgs? hahah
Validations on dot operators please :) , else your catch will throw errors

File: client/src/context/auth/AuthState.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+      loadUser();
+    } catch (err) {
+      dispatch({
+        type: LOGIN_FAIL,
+        payload: err.response.data.msg,
+      });
+    }
+  };
+
+  // Logout
```
Comment (rishabh1403) [inline client/src/context/auth/AuthState.js]: this comment is redundant. The function name is self-explanatory

File: client/src/context/auth/authReducer.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  USER_LOADED,
+  AUTH_ERROR,
+  LOGIN_SUCCESS,
+  LOGIN_FAIL,
+  LOGOUT,
+  CLEAR_ERRORS,
+} from '../types.js';
+
+export default (state, action) => {
+  switch (action.type) {
```
Comment (rishabh1403) [inline client/src/context/auth/authReducer.js]: add line breaks, unreadable code

**Review by rishabh1403 — CHANGES_REQUESTED** [review]
- remove redundant comments
- add validations
- add server url as constants
- add line breaks 

fix these things at all places

**Comment by p-racoon** [comment]
Lol 😂

**Comment by rishabh1403** [comment]
old memories

**Pull request closed without merging**